### DISCUSSION TITLE
Feature: Check for docker image updates and apply manual or automaticly

### DISCUSSION THREAD

**Discussion by Joooost** [IceWhaleTech/CasaOS#167] [Ideas]
#### Is your idea related to a problem? Please describe:
- To ease the maintenance of Docker containers it would save time and unnecessary steps to update installed Docker containers.

#### Describe the solution you'd like:
- As a user I want CasaOS periodically check for new (tag:latest) Docker Images for installed containers and show per container if there is an update available. So i can choose to update containers by a simple click or automatically if this is defined per installed container

- To support this feature it would be helpful to register some settings;
- Update check frequency
- Schedule (cronjob) to set the time for applying updates
- Option for containers to automatic or manual update the container

#### Describe alternatives you've considered:
- Running WatchTower container but it requires additional installation of a container

**Comment by rmcolbert** [comment]
There is already an open source docker container that can do this: https://www.dockupdater.dev/

This could be added as a pre-installed AppStore image (like Syncthin, Jellyfin or PhotoPrism)

**Reply by Joooost** [reply]
Nice, but should WatchTower be a better alternative as stated in the considerations? Is dockercontainer project still active? Last release is from 2 years ago. Is there a major difference between the two?

Love the suggestion of pre installing it, even beter would be to have an UI to add/change parameters and/or setup notifications.

**Reply by rmcolbert** [reply]
Here is a simple app store json file for Watchtower that can be used doing a custom install w/ import.
[Watchtower.zip](https://github.com/IceWhaleTech/CasaOS/files/8702147/Watchtower.zip)

Here is the file content for those who don't trust downloads :)

`{
    "custom_id": "ee93f94f-3589-4260-839e-39354834dc46",
    "origin": "custom",
    "network_model": "bridge",
    "index": "",
    "icon": "https://cdn.jsdelivr.net/gh/IceWhaleTech/AppIcon@main/all/watchtower.png",
    "image": "containrrr/watchtower:latest",
    "envs": [
        {
            "container": "PATH",
            "host": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "desc": "A container-based solution for automating Docker container base image updates.",
            "type": 0
        }
    ],
    "ports": [],
    "volumes": [
        {
            "container": "/var/run/docker.sock",
            "host": "/var/run/docker.sock",
            "type": 0,
            "desc": ""
        }
    ],
    "devices": [],
    "port_map": null,
    "cpu_shares": 10,
    "restart": "always",
    "enable_upnp": false,
    "label": "Watchtower",
    "description": "",
    "position": false,
    "host_name": "Watchtower",
    "privileged": false,
    "cap_add": [],
    "cmd": [],
    "version": "1.0"
}`
